### ISSUE TITLE
Shorthand -p (--for-release-of-packages) does not see vendored directories

### ISSUE THREAD

**Issue by prometheansacrifice** [ocaml/dune#2982]
Seen in ocaml-lsp: https://github.com/ocaml/ocaml-lsp/blob/master/ocaml-lsp-server.opam#L29
With the -p dune doesn't see the vendored merlin

**Comment by mseri** [comment]
I wonder how should dune differentiate vendored or non-vendored packages. If I have a multi-package repository, this is the behaviour I would expect in some sense. 

Should there be a way to explicitly specify a vendored packages folder and not have it implicit?

**Comment by andreypopp** [comment]
See linked PR, the invocation I'm using there is:
```
dune build
  --root .
  --ignore-promoted-rules
  --promote-install-files
  --no-config
  --profile release
  "%{name}%.install"
```
which does (hopefully) the same as:
```
dune build -p "%{name}%" @install
```
but doesn't not filter out rules in the workspace which belongs to packagfes other than `name` thus it allows to see and build vendored packages as well.

**Comment by diml** [comment]
Since 1.11 we can mark vendored subdirs explicitly via `(vendored_subdirs ...)`. So we could decide that vendored subdirs are not filtered out by `-p`. That seems reasonable: if you vendor something then you don't expect it to be a pre-built dependency.

Thoughts?

**Comment by mseri** [comment]
Sounds like a good idea to me

**Comment by rgrinberg** [comment]
@diml I think the `--release` flag fixes this issue?

**Comment by diml** [comment]
I'd say this is fixed since 2.2 because of #3705.
